p2p/transport/webtransport/transport.go

```diff
@@ -295,10 +295,13 @@ func (t *transport) CanDial(addr ma.Multiaddr) bool {
 }
 
 func (t *transport) Listen(laddr ma.Multiaddr) (tpt.Listener, error) {
-	isWebTransport, _ := IsWebtransportMultiaddr(laddr)
+	isWebTransport, certhashCount := IsWebtransportMultiaddr(laddr)
 	if !isWebTransport {
 		return nil, fmt.Errorf("cannot listen on non-WebTransport addr: %s", laddr)
 	}
+	if certhashCount > 0 {
+		return nil, fmt.Errorf("cannot listen on a specific certhash non-WebTransport addr: %s", laddr)
+	}
```

Comment:
Thanks! 😅

Reply:
Not strictly necessary for this PR, I should undo this and open it in a separate PR.

Reply:
oh no. It is the right thing. Plus a minor change. I think it is fine as is.

Reply:
#2426